From 607ff7dda35042ce1efdc2e131e8f3863d1e8bb5 Mon Sep 17 00:00:00 2001
From: Garrett Brown <eigendebugger@gmail.com>
Date: Tue, 1 Apr 2025 13:36:35 -0700
Subject: Fix Arduino toolchain compatibility with CMake 4.0

Update CMake policy minimum from 3.0 to 3.5 to resolve compatibility
errors after upgrading to CMake 4.0.
---
 Arduino-toolchain.cmake                            | 4 ++--
 Arduino/Templates/Scripts/BuildRuleScript.cmake.in | 2 +-
 Arduino/Templates/Scripts/LinkScript.cmake.in      | 2 +-
 Arduino/Templates/Scripts/ToolScript.cmake.in      | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Arduino-toolchain.cmake b/Arduino-toolchain.cmake
index 10634c0..bf33f09 100644
--- a/Arduino-toolchain.cmake
+++ b/Arduino-toolchain.cmake
@@ -33,8 +33,8 @@ endif()
 # Save the policy state. We will restore it at the end.
 cmake_policy(PUSH)
 
-# Set policy to above 3.0.0
-cmake_policy(VERSION 3.0.0)
+# Set policy to above 3.5.0
+cmake_policy(VERSION 3.5...4.0)
 
 # Interpret if() arguments without quotes as variables/keywords
 if (NOT CMAKE_VERSION VERSION_LESS 3.1)
diff --git a/Arduino/Templates/Scripts/BuildRuleScript.cmake.in b/Arduino/Templates/Scripts/BuildRuleScript.cmake.in
index 0ea16e4..ef6c445 100644
--- a/Arduino/Templates/Scripts/BuildRuleScript.cmake.in
+++ b/Arduino/Templates/Scripts/BuildRuleScript.cmake.in
@@ -1,6 +1,6 @@
 # Copyright (c) 2020 Arduino CMake Toolchain
 
-cmake_policy(VERSION 3.0)
+cmake_policy(VERSION 3.5...4.0)
 
 set(ARDUINO_TOOLCHAIN_DIR "@ARDUINO_TOOLCHAIN_DIR@")
 set(ARDUINO_PATTERN_NAMES 
diff --git a/Arduino/Templates/Scripts/LinkScript.cmake.in b/Arduino/Templates/Scripts/LinkScript.cmake.in
index bfe4822..cc76304 100644
--- a/Arduino/Templates/Scripts/LinkScript.cmake.in
+++ b/Arduino/Templates/Scripts/LinkScript.cmake.in
@@ -1,6 +1,6 @@
 # Copyright (c) 2020 Arduino CMake Toolchain
 
-cmake_policy(VERSION 3.0)
+cmake_policy(VERSION 3.5...4.0)
 
 # Generated content based on the selected board
 set(ARDUINO_TOOLCHAIN_DIR "@ARDUINO_TOOLCHAIN_DIR@")
diff --git a/Arduino/Templates/Scripts/ToolScript.cmake.in b/Arduino/Templates/Scripts/ToolScript.cmake.in
index 499c2c9..5d1a187 100644
--- a/Arduino/Templates/Scripts/ToolScript.cmake.in
+++ b/Arduino/Templates/Scripts/ToolScript.cmake.in
@@ -1,6 +1,6 @@
 # Copyright (c) 2020 Arduino CMake Toolchain
 
-cmake_policy(VERSION 3.1)
+cmake_policy(VERSION 3.5...4.0)
 
 # Generated content based on the selected board
 set(ARDUINO_TOOLCHAIN_DIR "@ARDUINO_TOOLCHAIN_DIR@")
-- 
2.43.0

